src/main/java/com/github/mc1arke/sonarqube/plugin/ce/pullrequest/bitbucket/server/BitbucketServerPullRequestDecorator.java

```diff
@@ -0,0 +1,439 @@
+        if (!optionalAnalysis.isPresent()) {
+            LOGGER.warn(
+                    "No analysis results were created for this project analysis. This is likely to be due to an earlier failure");
+            return;
+        }
+
+        Analysis analysis = optionalAnalysis.get();
+
+        Optional<String> revision = analysis.getRevision();
+        if (!revision.isPresent()) {
+            LOGGER.warn("No commit details were submitted with this analysis. Check the project is committed to Git");
+            return;
+        }
+
+        if (null == projectAnalysis.getQualityGate()) {
+            LOGGER.warn("No quality gate was found on the analysis, so no results are available");
+            return;
+        }
+
+        try {
+            Configuration configuration = configurationRepository.getConfiguration();
+            final String hostURL = getMandatoryProperty("sonar.pullrequest.bitbucket.url", configuration);
+            final String apiToken = getMandatoryProperty("sonar.pullrequest.bitbucket.token", configuration);
+            final String repositorySlug = getMandatoryProperty("sonar.pullrequest.bitbucket.repositorySlug", configuration);
+            final String pullRequestId = projectAnalysis.getBranch().flatMap(Branch::getName).get();
+            final String userSlug = configuration.get("sonar.pullrequest.bitbucket.userSlug").orElse(StringUtils.EMPTY);
+            final String projectKey = configuration.get("sonar.pullrequest.bitbucket.projectKey").orElse(StringUtils.EMPTY);
+
+            final boolean summaryCommentEnabled = Boolean.parseBoolean(getMandatoryProperty("sonar.pullrequest.summary.comment.enabled", configuration));
+            final boolean fileCommentEnabled = Boolean.parseBoolean(getMandatoryProperty("sonar.pullrequest.file.comment.enabled", configuration));
+            final boolean deleteCommentsEnabled = Boolean.parseBoolean(getMandatoryProperty("sonar.pullrequest.delete.comments.enabled", configuration));
+
+            final String commentUrl;
+            final String activityUrl;
+            final String diffUrl;
+            if (StringUtils.isNotBlank(userSlug)) {
+                commentUrl = String.format(FULL_PR_COMMENT_USER_API, hostURL, userSlug, repositorySlug, pullRequestId);
+                diffUrl = String.format(FULL_PR_DIFF_USER_API, hostURL, userSlug, repositorySlug, pullRequestId);
+                activityUrl = String.format(FULL_PR_ACTIVITIES_API, hostURL, userSlug, repositorySlug, pullRequestId, 250);
+            }
+            else if (StringUtils.isNotBlank(projectKey)) {
+                commentUrl = String.format(FULL_PR_COMMENT_API, hostURL, repositorySlug, projectKey, pullRequestId);
+                diffUrl = String.format(FULL_PR_DIFF_API, hostURL, repositorySlug, projectKey, pullRequestId);
+                activityUrl = String.format(FULL_PR_ACTIVITIES_USER_API, hostURL, repositorySlug, projectKey, pullRequestId, 250);
```

Comment:
The order of repositorySlug and projectKey is inverted here. Should be projectKey first.